apps/glusterfs/app_middleware.go

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright (c) 2017 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"net/http"
+	"strings"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/gorilla/context"
+	"github.com/urfave/negroni"
+
+	"github.com/heketi/heketi/pkg/kubernetes"
+)
+
+var (
+	kubeBackupDbToSecret = kubernetes.KubeBackupDbToSecret
+)
+
+// Authorization function
+func (a *App) Auth(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+
+	// Value saved by the JWT middleware.
+	data := context.Get(r, "jwt")
+
+	// Need to change from interface{} to the jwt.Token type
+	token := data.(*jwt.Token)
+	claims := token.Claims.(jwt.MapClaims)
+
+	// Check access
+	if "user" == claims["iss"] && r.URL.Path != "/volumes" {
+		http.Error(w, "Administrator access required", http.StatusUnauthorized)
+		return
+	}
+
+	// Everything is clean
+	next(w, r)
+}
+
+// Backup database to a secret
+func (a *App) BackupToKubernetesSecret(
+	w http.ResponseWriter,
+	r *http.Request,
+	next http.HandlerFunc) {
+
+	// Call the next middleware first
+	// Wrap it in a negroni ResponseWriter because for some reason
+	// the Golang http ResponseWriter does not provide access to
+	// the HttpStatus.
+	responsew := negroni.NewResponseWriter(w)
+	next(responsew, r)
+
+	// Only backup for POST and PUT
+	if !a.isAsyncDone(responsew, r) &&
+		r.Method == http.MethodGet {
+		return
+	}
```

Comment:
The indentation of the above section seems wrong, or at least somewhat misleading.